spring-cloud-gcp-pubsub/src/test/java/org/springframework/cloud/gcp/pubsub/integration/inbound/PubSubInboundChannelAdapterTests.java

```diff
@@ -31,13 +47,166 @@
 @RunWith(MockitoJUnitRunner.class)
 public class PubSubInboundChannelAdapterTests {
 
-	@Mock
-	private PubSubTemplate pubSubTemplate;
+	public static final String ACK = "ACK";
+
+	public static final String NACK = "NACK";
 
 	@Test(expected = IllegalArgumentException.class)
 	public void testNonNullAckMode() {
+		PubSubTemplate pubSubTemplate = Mockito.mock(PubSubTemplate.class);
+
 		PubSubInboundChannelAdapter adapter = new PubSubInboundChannelAdapter(
-				this.pubSubTemplate, "testSubscription");
+				pubSubTemplate, "testSubscription");
+
 		adapter.setAckMode(null);
 	}
+
+	@Test
+	public void testAckModeAuto() throws Exception {
+		PubSubTemplate pubSubTemplate = Mockito.mock(PubSubTemplate.class);
+		QueueChannel queueChannel = Mockito.mock(QueueChannel.class);
```

Comment:
I'm wondering what causes the send to this channel to always fail? Just because it's a mock?

Reply:
The problem that it is really an anti-pattern to mock against classes. Would be much better to do that against interfaces.
On the other hand I fully don't see reason to mock a `QueueChannel`: according the test body below, there is no any Mockito stubbing and verification. 🤷‍♂️